base/logging_test.go

```diff
@@ -242,3 +244,140 @@ func TestLogColor(t *testing.T) {
 	assert.Equal(t, "Format", color("Format", LevelTrace))
 	assert.Equal(t, "Format", color("Format", LevelNone))
 }
+
+func TestSGLevel(t *testing.T) {
+	assert.Equal(t, 1, DebugLevel.sgLevel())
+	assert.Equal(t, 1, InfoLevel.sgLevel())
+	assert.Equal(t, 2, WarnLevel.sgLevel())
+	assert.Equal(t, 2, ErrorLevel.sgLevel())
+	assert.Equal(t, 3, PanicLevel.sgLevel())
+	assert.Equal(t, 3, FatalLevel.sgLevel())
+	assert.Equal(t, 4, Level(5).sgLevel())
+}
+
+func TestMarshalTextError(t *testing.T) {
+	var level *Level
+	bytes, err := level.MarshalText()
+	assert.Nil(t, bytes, "bytes should be nil")
+	assert.Error(t, err, "Can't marshal a nil *Level to text")
+	assert.Equal(t, err.Error(), "can't marshal a nil *Level to text")
+}
+
+func TestGetCallersNameRecoverInfoImpossible(t *testing.T) {
+	callerName := GetCallersName(3, true)
+	assert.Equal(t, "???", callerName)
+	callerName = GetCallersName(3, false)
+	assert.Equal(t, "???", callerName)
+}
+
+func TestLastComponent(t *testing.T) {
+	path := lastComponent("/var/log/sync_gateway/sglogfile.log")
+	assert.Equal(t, "sglogfile.log", path)
+	path = lastComponent("\\var\\log\\sync_gateway\\sglogfile.log")
+	assert.Equal(t, "sglogfile.log", path)
+	path = lastComponent("sglogfile.log")
+	assert.Equal(t, "sglogfile.log", path)
+	path = lastComponent("/sglogfile.log")
+	assert.Equal(t, "sglogfile.log", path)
+	path = lastComponent("\\sglogfile.log")
+	assert.Equal(t, "sglogfile.log", path)
+}
+
+func TestLogSyncGatewayVersion(t *testing.T) {
+	out := CaptureOutput(t, LogSyncGatewayVersion)
+	assert.Contains(t, out, LongVersionString)
+	logger := &FileLogger{Enabled: true, logger: &log.Logger{}}
+
+	logger.level = LevelNone
+	assert.True(t, logger.shouldLog(LevelNone))
+	assert.False(t, logger.shouldLog(LevelWarn))
+	assert.False(t, logger.shouldLog(LevelError))
+	assert.False(t, logger.shouldLog(LevelInfo))
+	assert.False(t, logger.shouldLog(LevelDebug))
+	assert.False(t, logger.shouldLog(LevelTrace))
```

Comment:
The FileLogger above, and all of these `shouldLog` assertions are unnecessary and unrelated to this test. The code is already covered comprehensively by `TestFileShouldLog`.

The key piece we care about in the test is: "Does the console output contain stuff at every log level."

Reply:
This change has been made in the latest revision.